packages/split-button/src/Menu/Menu.tsx

```diff
@@ -74,58 +79,40 @@ export const Menu = ({
   useEventListener('keydown', handleKeyDown, { enabled: open });
   useBackdropClick(handleClose, [buttonRef, menuRef], open);
 
-  const renderMenuItems = () => {
-    const onChildClick = (
-      e: MouseEvent,
-      menuItem: ReactElement<MenuItemProps>,
-    ) => {
+  const renderMenuItems = useMemo(() => {
+    const onMenuItemClick = (e: MouseEvent, menuItem: MenuItemType) => {
       handleClose();
-      // @ts-expect-error - onClick is inferred from inferredPolymorphic
       menuItem.props.onClick?.(e);
+      onChange?.(e);
     };
 
-    const renderMenuItem = (
-      menuItem: ReactElement<MenuItemProps>,
-      index = 0,
-    ) => {
+    const renderMenuItem = (menuItem: MenuItemType) => {
       if (isComponentType(menuItem, 'MenuItem')) {
         return React.cloneElement(menuItem, {
           active: false,
-          key: `menuItem-${index}`,
-          // @ts-expect-error - onClick is inferred from inferredPolymorphic
-          onClick: (e: MouseEvent) => onChildClick(e, menuItem),
+          key: `menuItem-${menuItem.key}`,
+          onClick: (e: MouseEvent) => onMenuItemClick(e, menuItem),
         });
       } else {
-        console.warn('Please use a LeafyGreen <MenuItem> component.');
+        console.warn(
+          'Please use a LeafyGreen <MenuItem /> component. Received: ',
```

Comment:
People are probably going to want to use `SubMenu`, but we can deal with that later